projects/plugins/jetpack/_inc/client/components/contextualized-connection/index.jsx

```diff
@@ -0,0 +1,189 @@
+/**
+ * External dependencies
+ */
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { JetpackLogo } from '@automattic/jetpack-components';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { ConnectButton, ToS } from '@automattic/jetpack-connection';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * The Contextualized Connection component.
+ *
+ * @param {object} props -- The properties.
+ * @returns {React.Component} The `ContextualizedConnection` component.
+ */
+const ContextualizedConnection = props => {
```

Comment:
I'd consider giving this component a less generic name, since it's all about Jetpack + Woo at this state.
Something like `WooJetpackWelcomeScreen` maybe?

This is only a suggestion however, definitely not a blocker!

Reply:
>I'd consider giving this component a less generic name, since it's all about Jetpack + Woo at this state.
Something like WooJetpackWelcomeScreen maybe?

On the other hand, right now this component doesn't have any content related to Woo. All the Woo content comes from the props, that's why I chose a more generic name.